src/adaPets.js

```diff
@@ -9,29 +9,49 @@ const BASE_URL = 'http://localhost:3000/pets/';
 
 // Option functions.
 const listPets = () => {
-  // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch((error) => {
+      setError(`Unsuccessful get request.`);
+    });
 };
 
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to show details for a pet without selecting it!");
-    
   }
-
-  // Fill out as part of Wave 2.
+  axios.get(BASE_URL + selectedPetId)
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch(() => {
+      setError(`Request failed with status code 404 and failed to show details and select pet`);
+    });
```

Comment:
You should probably have made sure the status was actually 404 here:

```suggestion
    .catch((error) => {
      setError(`Request failed with status code ${error.response.status} and failed to show details and select pet`);
    });
```